lib/connect-backbone-to-react.js

```diff
@@ -124,11 +130,25 @@ module.exports = function connectBackboneToReact(
         // Bind event listeners for each model that changed.
         Object.keys(this.models).forEach(mapKey => {
           const model = this.models[mapKey];
-          if ((this.props.models && this.props.models[mapKey] === this.models[mapKey]) ||
-            (this.context.models && this.context.models[mapKey] === this.models[mapKey])) {
-            return; // Did not change.
+
+          // Retrieve old versions of the model from props and context for comparison.
+          const propsModel = this.props.models ? this.props.models[mapKey] : undefined;
+          const contextModel = this.context.models ? this.context.models[mapKey] : undefined;
+
+          // Do not attempt to create event listeners on an undefined model.
+          if (!model) {
+            // Instead, if it was previously defined, remove the old listeners.
+            if (propsModel) {
+              this.removeEventListener(mapKey, propsModel);
+            } else if (contextModel) {
+              // this.setModel prefers props to context, so we do that here too.
```

Comment:
Outdated comment?

Reply:
Ah no -- the comment refers to the `else if`. It's an attempt to explain the potentially surprising behavior that if a model with the matching mapKey exists in both `props` and `context`, we only remove listeners from the one in `props`. This is because only the one in `props` is actually used in `this.models`, per the `Object.assign` in setModel above.

Let me know if you think there's a cleaner way to express this?

Reply:
Ah.....how about lets use your wording? Put this above the if statement.

>  if a model with the matching mapKey exists in both props and context, we only remove listeners from the one in props. 
> We do this because only the one in props is actually used in this.models, per the Object.assign in setModel.